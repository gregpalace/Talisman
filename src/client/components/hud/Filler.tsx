import React from 'react';

interface IFiller {
  percentage: number | null;
}

const Filler = (props: IFiller) => {
  return <div className='filler' style={{ width: `${props.percentage}%` }} />;
};

export default Filler;
