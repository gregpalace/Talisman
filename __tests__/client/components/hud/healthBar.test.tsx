import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import HealthBar from '../../../../src/client/components/hud/healthBar';
import Filler from '../../../../src/client/components/hud/Filler';
import { wrap } from 'module';

describe('React unit tests', () => {
  let wrapper: any;

  const props = {
    percentage: 50,
  };
  beforeAll(() => {
    wrapper = shallow(<HealthBar {...props} />);
  });

  describe('HealthBar />', () => {
    it('should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render a <div/> with className of health-bar', () => {
      expect(wrapper.find('div').prop('className')).toBe('health-bar');
    });
    it('should render a <Filler /> component', () => {
      expect(wrapper.find(Filler)).toBeDefined();
      expect(wrapper.find(Filler).prop('percentage')).toBe(50);
    });
  });
});
