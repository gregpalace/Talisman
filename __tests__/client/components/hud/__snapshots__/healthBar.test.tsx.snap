// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests HealthBar /> should render correctly 1`] = `
<div
  className="health-bar"
>
  <Filler
    hp={100}
    percentage={50}
  />
</div>
`;
